tinygrad/tensor.py

```diff
@@ -2073,7 +2074,27 @@ def max_pool2d(self, kernel_size=(2,2), stride=None, dilation=1, padding=0, ceil
     """
     k_ = make_tuple(kernel_size, 2)
     pads = self._ceil_mode_padding2d(k_, stride if stride is not None else k_, dilation, padding) if ceil_mode else self._padding2d(padding, len(k_))
-    return self.pad(pads, value=dtypes.min(self.dtype))._pool(k_, stride if stride is not None else k_, dilation).max(tuple(range(-len(k_), 0)))
+    pooled = self.pad(pads, value=dtypes.min(self.dtype))._pool(k_, stride if stride is not None else k_, dilation)
+    ret = pooled.max(tuple(range(-len(k_), 0)))
+    if not return_indices: return ret
+    # HACK: add +1 offset to indices for `masked = (ret == flat_pool).where(flat_indices, 0)` to work with no 0's in `flat_indices`
+    indices = Tensor.arange(1, self.numel() + 1, dtype=self.dtype, device=self.device, requires_grad=False).reshape(self.shape)
+    indices = indices.pad(pads, value=dtypes.min(self.dtype))._pool(k_, stride if stride is not None else k_, dilation)
+    flat_indices, flat_pool = indices.flatten(-len(k_)), pooled.flatten(-len(k_))
+    # TODO: this contiguous changes result
```

Comment:
this implementation is a mess
but with this messy implementation I found another `.contiguous()` bug
result is wrong without this `contiguous`

maybe we do need a contiguous fuzzer 🤔